.github/workflows/build_docker_image.yml

```diff
@@ -16,76 +16,55 @@ jobs:
   build:
     name: Build
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        platform:
-          - amd64
-          - arm64
-          - arm/v7
-          - 386
     steps:
       - name: Checkout code from Git repository
         uses: actions/checkout@v2
 
-      - id: buildx
-        name: Set up Docker Buildx
-        uses: crazy-max/ghaction-docker-buildx@v1
-        with:
-          buildx-version: latest
-          qemu-version: latest
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
 
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKER_USERNAME }}
-          password: ${{ secrets.DOCKER_PASSWORD }}
+      # Only if you want the image to be push to docker.io instead of the GitHub Package repository
+#       - name: Docker login to DockerHub
+#         uses: docker/login-action@v1
+#         with:
+#           username: ${{ secrets.DOCKER_USERNAME }}
+#           password: ${{ secrets.DOCKER_PASSWORD }}
 
-      - name: Build & Push
-        run: |
-          IMAGE_ID=billzimmerman/raspap
-          # Change all uppercase to lowercase
-          IMAGE_ID=$(echo ${IMAGE_ID} | tr '[A-Z]' '[a-z]')
-          # Strip git ref prefix from version
-          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
-          # Strip "v" prefix from tag name
-          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo ${VERSION} | sed -e 's/^v//')
-          # Use Docker `latest` tag convention
-          [ "$VERSION" == "master" ] && VERSION=latest
-          if [ -n "${{ matrix.platform }}" ]; then
-            _platform=${{ matrix.platform }}
-            VERSION="${VERSION}-${_platform//\/}"
-          fi
-          echo "IMAGE_ID=${IMAGE_ID}"
-          echo "VERSION=${VERSION}"
-
-          docker buildx build \
-            --platform linux/${{ matrix.platform }} \
-            --tag ${IMAGE_ID}:${VERSION} \
-            --file ./Dockerfile \
-            --output type=image,push=true .
+      # Comment this if you prefer to use the docker.io image repository | This method can be better because it didn't need any password (use the credentials of the people who commit)
+      - name: Docker login to GitHub
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u "${{ github.actor }}" --password-stdin
```

Comment:
```suggestion
      - name: Login to DockerHub
        uses: docker/login-action@v1 
        with:
          registry: ghcr.io
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
Yep! Makes it a tad bit cleaner and we're using the official actions.